lib/challenge.rb

```diff
@@ -1,7 +1,18 @@
 # frozen_string_literal: true
 
 class FibonacciRange
-end
+  include Enumerable
+  attr_reader :stop
+
+  def initialize(stop)
+  @stop = stop
+  end
 
+    def each
+      ## starts at 0
+    end
```

Comment:
This is a really strong start! Good job!

Please review the solution to see how to complete it. 🙂 

```rb

class FibonacciRange
  include Enumerable
  attr_reader :stop
  private :stop

  def initialize(stop)
    @stop = stop
  end

  def each
    current = 0
    successor = 1
    while current < stop
      yield current
      current, successor = successor, current + successor
    end
    self
  end
end
```